### PR TITLE
added top 10 score

### PR DESCRIPTION
**Pull request by 0x-bear**

### CODE REVIEW

**Comment by phated** [comment]
@0x-bear any reason you don't just pull from the leaderboard API like the homepage? https://github.com/darkforest-eth/client/blob/05c7731df52932455c6fffddd9512c194f9d49bd/src/Backend/Network/LeaderboardApi.ts#L5

File: content/utilities/top10/plugin.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  constructor() {}
+
+  /**
+   * Called when plugin is launched with the "run" button.
+   */
+  async render(container) {
+    container.style.width = "280px";
+    let label = document.createElement('div');
+        
+table=[];
```
Comment (jacobrosenthal) [inline content/utilities/top10/plugin.js:14]:
```suggestion
let table=[];
```

File: content/utilities/top10/plugin.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+players=df.getAllPlayers().sort((p1,p2)=>{if (p1.withdrawnSilver + p1.totalArtifactPoints > p2.withdrawnSilver + p2.totalArtifactPoints) { return -1; } else { return 1 }})
```
Comment (jacobrosenthal) [inline content/utilities/top10/plugin.js:15]:
```suggestion
let players=df.getAllPlayers().sort((p1,p2)=>{if (p1.withdrawnSilver + p1.totalArtifactPoints > p2.withdrawnSilver + p2.totalArtifactPoints) { return -1; } else { return 1 }})
```

File: content/utilities/top10/plugin.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+.slice(0,10)
+.forEach(function(item,index,array){
+  if (item.twitter == undefined) { name = item.address.substr(0,5); } else { name = item.twitter }
+  newRow = [name,Math.round(item.withdrawnSilver/1000),Math.round(item.totalArtifactPoints/1000),Math.round((item.withdrawnSilver+item.totalArtifactPoints)/1000)]
```
Comment (jacobrosenthal) [inline content/utilities/top10/plugin.js:19]:
```suggestion
  let newRow = [name,Math.round(item.withdrawnSilver/1000),Math.round(item.totalArtifactPoints/1000),Math.round((item.withdrawnSilver+item.totalArtifactPoints)/1000)]
```

File: content/utilities/top10/plugin.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+// top 10 pulled from ingame data - @0x_Bear
+console.log(df, ui);
```
Comment (jacobrosenthal) [inline content/utilities/top10/plugin.js:2]:
```suggestion
```

**Review by jacobrosenthal — CHANGES_REQUESTED** [review]
In previous rounds scoring was very expensive to calculate, but its not too bad for this round it seems so might be doable. 

Can you run a formatter across this code, its very hard to evaluate as is.

File: content/utilities/top10/index.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+title: Top10
```
Comment (jacobrosenthal) [inline content/utilities/top10/index.md:2]: maybe worth clarifying,since youre not using the official api
```suggestion
title: Unofficial Top10
```

**Comment by jacobrosenthal** [comment]
Actually yeah remembering now the game only loads planets from the contract that you 'touch'(ie have seen) so thats why we havent been able to do a scoring plugin outside of using a datasource that has all the planet information ie graphapi or possible in this case using the df server url (that is using thegraph api behind the scenes)

File: content/utilities/top10/index.md (hunk 5 above)
Comment (0x-bear) [reply content/utilities/top10/index.md:2]: very experimental, keep track of score at your own risk :)

**Comment by jacobrosenthal** [comment]
ping @0x-bear You wanna swap over to https://api.zkga.me/leaderboard  so itll work for next round?

**Comment by jacobrosenthal** [comment]
Closing for #116 :heart: @0x-bear

**Pull request closed without merging**